### PR TITLE
[newrelic-infrastructure] reduce scope of permissions

### PR DESCRIPTION
**Pull request by paologallinaharbur**
#### Is this a new chart
no
#### What this PR does / why we need it:
Reducing the scope of the cluster role binding that is currently too broad

#### Which issue this PR fixes
  - fixes https://github.com/newrelic/helm-charts/issues/282

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/newrelic-infrastructure/templates/rolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "newrelic.fullname" . }}-secrets
+subjects:
+- kind: ServiceAccount
+  name: {{ template "newrelic.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace }}
+{{- if .Values.etcdTlsSecretNamespace}}
```
Comment (roobre) [inline charts/newrelic-infrastructure/templates/rolebinding.yaml:17]: I don't get why these are two different cases. How are we going to scope the secret (`limitSecretAccess == true`) if `etcdTlsSecretNamespace` is not defined?

Couldn't we just replace the whole `limitSecretAccess` flag by a `etcdTlsSecretNamespace != ""` check?

Comment (paologallinaharbur) [reply charts/newrelic-infrastructure/templates/rolebinding.yaml:17]: You are right, the change is safe enough to be done without introducing a new flag.

By the way we should change as well the static manifests in the nri-kubernetes repo

File: charts/newrelic-infrastructure/templates/clusterrole.yaml [hunk 2]
```diff
@@ -27,4 +26,17 @@ rules:
...
+metadata:
+  labels: {{ include "newrelic.labels" . | indent 4 }}
+  name: {{ template "newrelic.fullname" . }}-secrets
+rules:
+  - apiGroups: [""]
+    resources:
+      - "secrets"
+    verbs: ["get", "list"]
 {{- end -}}
+{{- end -}}
```
Comment (roobre) [inline charts/newrelic-infrastructure/templates/clusterrole.yaml:42]: Newline before `EOF`

Comment (paologallinaharbur) [reply charts/newrelic-infrastructure/templates/clusterrole.yaml]: 🙅

**Review by roobre — APPROVED** [review]
If manual testing passes, lgtm

**Pull request merged**